examples/recursion.js

```diff
@@ -0,0 +1,14 @@
+var typed = require('../typed-function');
+
+// create a typed function that invokes itself
+var sqrt = typed({
+  'number': function (value) {
+    return Math.sqrt(value);
+  },
+  'string': function (value) {
+    return this.call(parseInt(value, 10));
+  }
+});
+
+// use the typed function
+console.log(sqrt(9));         // output: 3
```

Comment:
Maybe good to also demo `console.log(sqrt('9'));` here, passing a string as input 😄

Reply:
Haha, oops!  Good catch.